extensions/fairifier-rdf-extension/dtls/fairifier/PostFairDataToFairDataPoint.java

```diff
@@ -54,188 +34,268 @@
+            } else {
+                distributionMetadata.setParentURI(f.createIRI(fdp.getString("baseUri") + "/dataset/"
+                        + dataset.getString("http://purl.org/dc/terms/title").replace(" ", "_") + "_"
+                        + dataset.getString("http://purl.org/dc/terms/hasVersion").replace(" ", "_") + "_"
+                        + uuid_dataset));
             }
-            distributionMetadata.setVersion(f.createLiteral(distribution.getString("http://purl.org/dc/terms/hasVersion")) );
-//          optional
-            try{
-                distributionMetadata.setLicense(f.createIRI(distribution.getString("http://purl.org/dc/terms/license")));
-            }catch(Exception ex){}
-            distributionMetadata.setUri( f.createIRI( "http://base/catalog/dataset/distribution"));
+            distributionMetadata
+                    .setVersion(f.createLiteral(distribution.getString("http://purl.org/dc/terms/hasVersion")));
+            // optional
+            try {
+                distributionMetadata
+                        .setLicense(f.createIRI(distribution.getString("http://purl.org/dc/terms/license")));
+                distributionMetadata.setLicense(f.createIRI(distribution.getString("http://purl.org/dc/terms/rights")));
+            } catch (Exception ex) {
+                LOGGER.warn(ex.getMessage());
+            }
+            distributionMetadata.setUri(f.createIRI("http://base/catalog/dataset/distribution"));
             distributionMetadata.setIssued(f.createLiteral(date));
             distributionMetadata.setModified(f.createLiteral(date));
-//          optional
-            try{
-                distributionMetadata.setDescription(f.createLiteral(distribution.getString("http://purl.org/dc/terms/description")) );
-            }catch(Exception ex){}
-            distributionString = MetadataUtils.getString(distributionMetadata, RDFFormat.TURTLE,false).replaceAll("\\<" + distributionMetadata.getUri() + "\\>","<>");
+            // optional
+            try {
+                distributionMetadata.setDescription(
+                        f.createLiteral(distribution.getString("http://purl.org/dc/terms/description")));
+            } catch (Exception ex) {
+            }
+            distributionString = MetadataUtils.getString(distributionMetadata, RDFFormat.TURTLE, false)
+                    .replaceAll("\\<" + distributionMetadata.getUri() + "\\>", "<>");
             String catalogPost = null;
             String datasetPost = null;
-            if (!catalog.getBoolean("_exists")){
-                catalogPost = IOUtils.toString(HttpUtils.post(fdp.getString("baseUri") + "/catalog?id=" + catalog.getString("http://purl.org/dc/terms/title").replace(" ","_")+"_"+catalog.getString("http://purl.org/dc/terms/hasVersion").replace(" ","_")+"_"+uuid_catalog, catalogString, "text/turtle").getContent(), "UTF-8");
+            if (!catalog.getBoolean("_exists")) {
+                catalogPost = IOUtils
```

Comment:
Too many things happen in 1 line here, no way to figure out what goes wrong when either reading it or communicating it back to the front-end.